DNN Platform/DotNetNuke.Web/Common/RequestScopeServiceProvider.cs

```diff
@@ -0,0 +1,75 @@
+﻿namespace DotNetNuke.Web.Common
+{
+    using System;
+    using System.Collections.Concurrent;
+    using System.Reflection;
+    using System.Web;
+
+    using DotNetNuke.Common;
+    using DotNetNuke.Common.Extensions;
+    using Microsoft.Extensions.DependencyInjection;
+
+    /// <summary>
+    /// Service provider implementation for <see cref="HttpRuntime.WebObjectActivator"/>.
+    /// </summary>
+    internal class RequestScopeServiceProvider : IServiceProvider
+    {
+        private const int TypesCannotResolveCacheCap = 100000;
```

Comment:
Is there a particular reason to limit this?  Just thinking if not necessary we could remove some operations/calcuation etc

Reply:
The only reason I can think of is whenever there is a service that is using a generic definition (e.g. `SomeControl<T>`). Every type (e.g. `SomeControl<int>` or `SomeControl<bool>`) will be added to this dictionary.

I've added this check because the Unity implementation (see https://github.com/aspnet/AspNetWebFormsDependencyInjection/blob/master/src/UnityAdapter/ContainerServiceProvider.cs#L71, created by a Microsoft developer as example) did this as well.

Reply:
This appears to be used as a safe-guard in case the `RequestScopeServiceProvider` has already used the `Activator`. It tries to auto resolve the instance and not use the container. This is useful in the current implementation because it prevent performance degradation which is caused by the exception handling control flow.

The CacheCap is just a magic number to prevent the container from using up way too much memory. If we try to resolve that many objects that aren't on the container we are doing something wrong. The application shouldn't necessarily crash at that point, but it will go through the entire `GetService()` code on every request even if the object isn't in the container.

I don't believe the cache dictionary is needed, see my comment's above https://github.com/dnnsoftware/Dnn.Platform/pull/4098#discussion_r491128632

Reply:
I've done some research with performance in https://github.com/dnnsoftware/Dnn.Platform/pull/4098#discussion_r491345847. Here we can see that using `Type.GetConstructors()` is faster than using a cache dictionary, so we can remove the dictionary (and the limit).